src/python/pygrackle/utilities/model_tests.py

```diff
@@ -135,22 +138,139 @@ def generate_model_sets(model_store):
+        "--skip-plot", action="store_true", help="skip any plotting"
+    )
+
+    _triples = [
+        ("show-param-preset", "show the parameter preset", True),
+        ("show-input-preset", "show the input-grid preset", True),
+        ("num-param-preset", "number of parameter presets", False),
+        ("num-input-preset", "number of input-grid presets", False),
+    ]
+    for name, descr, has_arg in _triples:
+        p = subparsers.add_parser(name, description=descr)
+        p.set_defaults(kind=name)
+        if has_arg:
+            p.add_argument("index", type=int, help="index of the preset")
+    return parser
+
+
+_MODEL_CLI_PARSER = _build_base_parser()
+
+class ParsedModelProps(NamedTuple):
+    my_chemistry: chemistry_data
+    input_set: dict
+    skip_plot: bool
+    output_name: str
+    extra_attrs: dict
+
+def _get_preset(index_value, sets, index_name):
+    try:
+        return sets[index_value]
+    except IndexError:
+        if len(sets) == 1:
+            raise ValueError(f"{index_name} must be 0") from None
+        raise ValueError(
+            f"{index_name} must satisfy 0 <= index < {len(sets)}"
+        ) from None
+
+def parse_model_cliargs(arg_l, model_name):
     """
+    Parses command line arguments and takes appropriate actions
 
-    if model_name not in model_sets:
-        raise ValueError("Unkown model name: {model_name}.")
+    Parameters
+    ----------
+    arg_l: list of str
+        The list of cli args
+    model_name: str
+        The name of the model to be tested
 
+    Returns
+    -------
+    out : None or ParsedModelProps
+        If a subcommand was invoked that implies that the program should
+        immediately exit, this returns None. Otherwise, this provides the
+        appropriate data to be used while invoking the simple model's logic
+
+    Notes
+    -----
+    Rather than defining the _model_test_grids variable inside of this file and
+    matching up the keys with specified model_name argument, I think it might
+    be more robust for the possible _model_test_grids to be directly passed
+    into this function.
+    """
+    if model_name not in model_sets:
+        raise ValueError(f"Unkown model name: {model_name}.")
     my_model = model_sets[model_name]
-    par_set = my_model["parameter_sets"][parameter_index]
-    input_set = my_model["input_sets"][input_index]
-
-    my_chemistry = chemistry_data()
-    for par, val in par_set.items():
-        if par == "grackle_data_file":
-            val = os.path.join(grackle_data_dir, val)
-        setattr(my_chemistry, par, val)
-    return my_chemistry, input_set
+    par_sets, input_sets = my_model["parameter_sets"], my_model["input_sets"]
+
+    args = _MODEL_CLI_PARSER.parse_args(arg_l)
+
+    if args.kind == "num-param-preset":
+        print(len(par_sets))
+        return None
+    elif args.kind == "num-input-preset":
+        print(len(input_sets))
+        return None
+    elif args.kind == "show-param-preset":
```

Comment:
(Needed if we make the above simplification)

```suggestion
    if args.kind == "show-param-preset":
```